### DISCUSSION TITLE
XpMeter

### DISCUSSION THREAD

**Discussion by jamesbellamy99** [whs/runekit#15] [Q&A macOS]
Currently the XpMeter seems unable to detect my metrics panel. 
Any recommendations for trouble shooting?

**Comment by emiliyo9** [comment]
I have the same problem not being able to detect the metrics panel. Taking a look at the logs I get the following error:
`ERROR:runekit.app.view.browser_window.PageClass.https://runeapps.org/apps/alt1/xpmeter/:Uncaught NetworkError: Failed to execute 'send' on 'XMLHttpRequest': Failed to load 'rk:{"func":"getRegionRaw","x":0,"y":0,"w":1920,"h":1034}'.`
Maybe we can go on from here.

**Reply by emiliyo9** [reply]
ps: This is on a dell G5 running manjaro linux.

**Comment by whs** [comment]
As listed in [troubleshooting](https://github.com/whs/runekit/wiki/Troubleshooting#xp-counter-in-afkscape-is-not-working), the XP counter is not supported as it does the OCR on Alt1 host application.

I had several attempt at this, but preexisting Python image processing libraries is not as fast as Alt1's (as it is optimized just for RuneScape, not just any image) and would make the app hang every other second. Porting Alt1's JavaScript code is also hard as the code is quite messy (even the author think so themselves). If anyone want to give it a go, the `whs/ocr` have my work in progress committed

**Reply by schoentoon** [reply]
https://github.com/whs/ocr seems to give a 404 to me, did you leave it private by any chance?

**Reply by whs** [reply]
Oh, I forgot to write "branch" - https://github.com/whs/runekit/tree/whs/ocr
